scripts/setup-helper-functions.sh

```diff
@@ -156,29 +159,50 @@ function get_cxx_flags {
 function wget_and_untar {
   local URL=$1
   local DIR=$2
+  mkdir -p "${DEPENDENCY_DIR}"
+  pushd "${DEPENDENCY_DIR}"
+  SUDO="${SUDO:-""}"
+  if [ -d "${DIR}" ]; then
+    if prompt "${DIR} already exists. Delete?"; then
+      ${SUDO} rm -rf "${DIR}"
+    else
+      popd
+      return
+    fi
+  fi
   mkdir -p "${DIR}"
   pushd "${DIR}"
   curl -L "${URL}" > $2.tar.gz
   tar -xz --strip-components=1 -f $2.tar.gz
   popd
+  popd
 }
 
-function cmake_install {
-  if [ -d "$1" ]; then
-    DIR="$1"
-    shift
-  else
-    DIR=$(pwd)
-  fi
+function cmake_install_dir {
+  pushd "${DEPENDENCY_DIR}/$1"
   local NAME=$(basename "$(pwd)")
   local BINARY_DIR=_build
   SUDO="${SUDO:-""}"
-  pushd "${DIR}"
-  if [ -d "${BINARY_DIR}" ] && prompt "Do you want to rebuild ${NAME}?"; then
-    ${SUDO} rm -rf "${BINARY_DIR}"
+  if [ -d "${BINARY_DIR}" ]; then
+    if prompt "Do you want to rebuild ${NAME}?"; then
```

Comment:
Since you are updating `cmake_install`, please fix `${NAME}` per #10908, and then I will close my PR. Thanks!

Sth like
```sh
local DIR="${DEPENDENCY_DIR}/$1"
local NAME=${basename "${DIR}"}
pushd "${DIR}"
```

Reply:
My PR fixes this by moving `pushd $DIR` above NAME.
Your change is not really needed.